back-end/src/sciler/handler/handler.go

```diff
@@ -89,36 +90,47 @@ func (handler *Handler) onConnectionMsg(raw Message) {
 
 //onStatusMsg is the function to process status messages.
 func (handler *Handler) onStatusMsg(raw Message) {
+	contents := raw.Contents.(map[string]interface{})
 	_, ok := handler.Config.Devices[raw.DeviceID]
 	if !ok {
 		logrus.Warn("status message received from device " + raw.DeviceID + ", which is not in the config")
 	} else {
 		logrus.Info("status message received from: " + raw.DeviceID + ", status: " + fmt.Sprint(raw.Contents))
-		for k, v := range raw.Contents {
+		for k, v := range contents {
 			handler.Config.Devices[raw.DeviceID].Status[k] = v
 		}
 	}
 }
 
 //onConfirmationMsg is the function to process confirmation messages.
 func (handler *Handler) onConfirmationMsg(raw Message) {
-	value, ok := raw.Contents["completed"]
+	contents := raw.Contents.(map[string]interface{})
+	value, ok := contents["completed"]
 	if !ok || reflect.TypeOf(value).Kind() != reflect.Bool {
 		logrus.Error("received improperly structured confirmation message from device " + raw.DeviceID)
 	} else {
-		original, ok := raw.Contents["instructed"]
+		original, ok := contents["instructed"]
 		if !ok {
 			logrus.Error("received improperly structured confirmation message from device " + raw.DeviceID)
 		} else {
 			msg := original.(map[string]interface{})
+			contents, err := getMapSlice(msg["contents"])
+			if err != nil {
+				logrus.Error(err)
+				return
+			}
+
+			var instructionString string
+			for i, instruction := range contents {
+				instructionString = fmt.Sprintf("%s%d: %s ", instructionString, i, instruction["instruction"])
```

Comment:
should this be added to? it seems to be overwritten

Reply:
rewrote this with a +=, which is indeed nicer